v2/cmd/controller/logging/logging.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package logging
+
+import (
+	"flag"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/klog/v2/textlogger"
+)
+
+var (
+	// verbose indicates whether we should use verbose logging.
+	// Default is no
+	verbose *bool
+
+	// useJson indicates whether we should output logs in JSON format.
+	// Default is no
+	useJson *bool
+)
+
+// Create returns a new logger, ready for use.
+// This can be called multiple times if required.
+func Create() logr.Logger {
```

Comment:
This would also more closely mirror what both `TextLogger` and `zap` seem to do, which is take their configuration at the instance level rather than the package level.

Reply:
Done.